tests/qat/test_quantum_backend.py

```diff
@@ -362,6 +366,54 @@ def test_setup_hold_measure_pulse(self):
             np.round(np.imag(result[setup_length:full_length]), 7),
         )
 
+    @pytest.mark.parametrize(
+        "width, warning_flag",
+        [
+            (3.2e-08, False),
+            (5e-09, True),
+            (3.3e-08, True),
+            (15e-09, True),
+            (1.6e-08, False),
+        ],
+    )
+    def test_pulse_validation_grid_time_throws_warning(self, width, warning_flag, caplog):
+        hw = get_test_model()
+        qubit = hw.get_qubit(0)
+        drive_channel = qubit.get_drive_channel()
+
+        builder = get_builder(hw).pulse(drive_channel, PulseShapeType.SQUARE, width=width)
+
+        engine = get_test_execution_engine(hw)
+
+        # Capture if warnings are sent to logger.
+        with caplog.at_level(logging.WARNING):
+            execute_instructions(engine, builder)
+            assert (
+                "Non-hardware-feasible pulse detected" in caplog.text
+                and str(width) in caplog.text
+            ) == warning_flag
+
+    def test_pulse_validation_grid_time_throws_warning_for_sweep(self, caplog):
+        hw = get_test_model()
+        qubit = hw.get_qubit(0)
+        drive_channel = qubit.get_drive_channel()
+        engine = get_test_execution_engine(hw)
+
+        time = np.linspace(
+            4e-09, 1.6e-8, 11
+        )  # Contains non-hardware-feasible pulse widths.
+        print(time)
```

Comment:
log here instead of print ^^

Reply:
Oh that one was during testing that I forgot to remove afterwards :) thanks!